### ISSUE TITLE
Redirect to new page on successful form submission contains unexpected text/vnd.turbo-stream.html

### ISSUE THREAD

**Issue by pySilver** [hotwired/turbo#1018]
I'm surprised to see that form POST submission that responded with a redirect results is following this redirect using GET method with  `text/vnd.turbo-stream.html` header attached. Is that a bug or desired behaviour? It is really hard to distinguish such redirect from other stream requests on backend. I believe it's a bug because if I wanted to respond with a stream I'd do it directly, instead of sending redirect header.

**Comment by seanpdoyle** [comment]
@pySilver how was the original `POST` request submitted? Was it through a `<form>` element submission or a `fetch` request?

What is the status code in the response from the server? Is it a [303 See Other](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303) like the [Turbo Drive](https://turbo.hotwired.dev/handbook/drive#redirecting-after-a-form-submission) documentation suggests, or another `3xx` code?

The `303 See Other` redirect would result in a subsequent `GET` request, whereas other status codes would use the request's verb (in this case, a `POST`).

**Comment by pySilver** [comment]
@seanpdoyle Sorry for the delay with the reply. Setup was the following:

1. POST form with various input fields. Form **is not** nested in a frame.
2. Backend endpoint that consumes data and returns 302/303 redirect (does not matter) to some location within the same domain and protocol (actually it's localhost at the moment).

**What I've expected**: The user is redirected to a new location using GET request without any specific request headers attached. 
**Observed behavior**: The user is redirected -//- but Turbo accept header is being repeated for final location.

**Comment by seanpdoyle** [comment]
@pySilver thank you for clarifying!

I believe that the client-side of Turbo doesn't have access to the intermediate redirect response headers, so it doesn't have access to remove the header.

**Comment by pySilver** [comment]
@seanpdoyle  Yeah. This is what I thought - it's probably the browser that is following a redirect and applying previous headers automatically (or more specifically Fetch API). I'm wondering: How it differ when the form is wrapped in the frame?
